### PR TITLE
fix(issue_search): Fix bug with dynamic counts in issue search

### PR DESCRIPTION
**Pull request by wedamija**
This was missed in the original fix.

### CODE REVIEW

**Review by wmak — APPROVED** [review]

**Review by iProgramStuff — APPROVED** [review]
Thank you Dan!

File: src/sentry/api/serializers/models/group.py [hunk 1]
```diff
@@ -824,7 +823,9 @@ def get_attrs(self, item_list, user):
                 )
             for item in item_list:
                 if self.has_dynamic_issue_counts:
-                    attrs[item].update({"filtered_stats": filtered_stats[item.id]})
+                    attrs[item].update(
+                        {"filtered_stats": filtered_stats[item.id] if filtered_stats else {}}
```
Comment (taylangocmen) [inline src/sentry/api/serializers/models/group.py:827]: I think this should still be `"stats"` instead of `"filtered_stats"`

Comment (iProgramStuff) [reply src/sentry/api/serializers/models/group.py:827]: On line 844 "filtered_stats" is renamed to stats when brought into the "filtered" dict:
`attrs["filtered"].update({"stats": {self.stats_period: attrs["filtered_stats"]}})`

**Comment by wedamija** [comment]
https://github.com/getsentry/sentry/pull/20904

**Pull request closed without merging**